gopass ls of an unavailable entry causes nil pointer dereference

##Steps to reproduce

```
✗ gopass ls anentrynotthere
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x56541a]

goroutine 1 [running]:
github.com/justwatchcom/gopass/tree/simple.(*Folder).SetRoot(0x0, 0x7ffe20de4101)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/tree/simple/folder.go:238 +0xa
github.com/justwatchcom/gopass/action.(*Action).List(0xc420010de0, 0xc4200908c0, 0xc420052c38, 0xc420052c38)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/action/list.go:57 +0x4e8
github.com/justwatchcom/gopass/action.(*Action).List-fm(0xc4200908c0, 0x0, 0x0)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/main.go:369 +0x34
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.HandleAction(0x67ade0, 0xc4200192f0, 0xc4200908c0, 0xc42001a700, 0x0)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/app.go:485 +0xd4
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.Command.Run(0x6c67f4, 0x4, 0x0, 0x0, 0xc420019640, 0x1, 0x1, 0x6c8a43, 0xd, 0x0, ...)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/command.go:193 +0xb72
github.com/justwatchcom/gopass/vendor/github.com/urfave/cli.(*App).Run(0xc42009c9c0, 0xc420010270, 0x3, 0x3, 0x0, 0x0)
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/vendor/github.com/urfave/cli/app.go:250 +0x7d0
main.main()
	/home/zabilcm/go/src/github.com/justwatchcom/gopass/main.go:551 +0x4b6a
```
```
✔︎ sam (master) ✗ gopass --version
gopass HEAD (n/a ) go1.8.3

```
```
✗ uname -smovri
Linux 4.8.0-58-generic #63~16.04.1-Ubuntu SMP Mon Jun 26 18:08:51 UTC 2017 x86_64 x86_64 GNU/Linux

```

--- a/action/list.go
+++ b/action/list.go
@@ -49,8 +49,9 @@ func (s *Action) List(c *cli.Context) error {
 		return nil
 	}
 
-	subtree := l.FindFolder(filter)
-	if subtree == nil {
+	subtree, err := l.FindFolder(filter)
+	if err != nil {
+		color.Red("Entry '%s' not found", filter)
 		return nil
 	}
 

--- a/action/totp.go
+++ b/action/totp.go
@@ -64,7 +64,7 @@ func printCode(secret string, t time.Time) (string, error) {
 	}
 
 	expiresAt := time.Unix(t.Unix()+totpPeriod-(t.Unix()%totpPeriod), 0)
-	secondsLeft := int(expiresAt.Sub(time.Now()) / time.Second)
+	secondsLeft := int(time.Until(expiresAt).Seconds())
 
 	if secondsLeft <= totpPeriod {
 		color.Yellow("%s lasts %ds \t|%s%s|", code, secondsLeft, strings.Repeat("=", totpPeriod-secondsLeft), strings.Repeat("-", secondsLeft))

--- a/tree/simple/folder.go
+++ b/tree/simple/folder.go
@@ -167,8 +167,12 @@ func (f *Folder) getFolder(name string) *Folder {
 }
 
 // FindFolder returns a sub-tree or nil, if the subtree does not exist
-func (f *Folder) FindFolder(name string) tree.Tree {
-	return f.findFolder(strings.Split(strings.TrimSuffix(name, "/"), "/"))
+func (f *Folder) FindFolder(name string) (tree.Tree, error) {
+	sub := f.findFolder(strings.Split(strings.TrimSuffix(name, "/"), "/"))
+	if sub == nil {
+		return nil, fmt.Errorf("Entry not found")
+	}
+	return sub, nil
 }
 
 // findFolder recursively tries to find the named sub-folder

--- a/tree/simple/tree_test.go
+++ b/tree/simple/tree_test.go
@@ -78,8 +78,8 @@ func TestFormatSubtree(t *testing.T) {
 			t.Fatalf("failed to add file: %s", err)
 		}
 	}
-	sub := root.FindFolder("baz/ing")
-	if sub == nil {
+	sub, err := root.FindFolder("baz/ing")
+	if err != nil {
 		t.Fatalf("failed to find subtree")
 	}
 	got := strings.TrimSpace(sub.Format(0))
@@ -87,4 +87,24 @@ func TestFormatSubtree(t *testing.T) {
 	if want != got {
 		t.Errorf("Format mismatch: %s vs %s", want, got)
 	}
+}
+
+func TestGetNonExistingSubtree(t *testing.T) {
+	root := New("gopass")
+	for _, f := range []string{
+		"foo/bar",
+		"foo/baz",
+		"baz/ing/a",
+		"baz/ing/b",
+	} {
+		if err := root.AddFile(f, "text/plain"); err != nil {
+			t.Fatalf("failed to add file: %s", err)
+		}
+	}
+	sub, err := root.FindFolder("bla")
+	if err == nil {
+		t.Fatalf("should fail to find subtree")
+	}
+	// if it doesn't panic we're good
+	_ = sub
 }

--- a/tree/tree.go
+++ b/tree/tree.go
@@ -8,7 +8,7 @@ type Tree interface {
 	AddFile(string, string) error
 	AddMount(string, string) error
 	AddTemplate(string) error
-	FindFolder(string) Tree
+	FindFolder(string) (Tree, error)
 	SetRoot(bool)
 	SetName(string)
 	Len() int

